gradoop-flink/src/main/java/org/gradoop/flink/algorithms/fsm/dimspan/DIMSpan.java

```diff
@@ -353,14 +354,17 @@ public DIMSpan(DIMSpanConfig fsmConfig) {
    *
    * @return sum group combine function
    */
-  protected GroupCombineFunction<WithCount<int[]>, WithCount<int[]>>
-  sumPartition() {
+  private GroupCombineFunction<WithCount<int[]>, WithCount<int[]>> sumPartition() {
 
-    SumAggregationFunction.LongSumAgg[] sum = { new SumAggregationFunction.LongSumAgg() };
+    @SuppressWarnings("unchecked")
+    AggregationFunction<Long>[] sum = new AggregationFunction[] {
+      new SumAggregationFunction.SumAggregationFunctionFactory()
+        .createAggregationFunction(Long.class)
+    };
 
     int[] fields = { 1 };
 
-    return new AggregateOperator.AggregatingUdf(sum, fields);
+    return new AggregateMultipleFunctions(sum, fields);
   }
```

Comment:
@s1ck Basically AggregatingUdf was public in previous versions of Flink, now its private while the functionality remains the same. At this point we just need the function of AggregatingUdf. So AggregateOperator does a lot of stuff we don't need here. I re-implemented the function we need but still i kinda dislike this option.

Reply:
I was wondering why don't implement `AggregateMultipleFunction` generic?

Reply:
One thing you could do instead is using reflection to create an instance of the inner class. See https://stackoverflow.com/questions/15015675/accessing-non-visible-classes-with-reflection

Reply:
But since it's only used within Dimspan, a specific implementation as you did it is also fine. If there is at any time the need to reuse that for another aggregation function, one could invest more engineering.

Reply:
Using reflection for that particular case seems kinda hacky. I'll follow your intention and if the aggregation function needs to be used by other algorithms then dimspan we should invest more engineering.